src/components/hooks/useContributionsReminder.ts

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
+import { addContributionReminderCookie, ReminderStatus } from '../modules/utils/reminders';
+import {
+    OneOffSignupRequest,
+    ReminderPlatform,
+    ReminderComponent,
+    ReminderStage,
+} from '../../api/supportRemindersApi';
+
+const CREATE_ONE_OFF_REMINDER_ENDPOINT = 'https://support.theguardian.com/reminders/create/one-off';
+
+interface ContributionsReminder {
+    reminderStatus: ReminderStatus;
+    createReminder: (email: string) => void;
+}
+
+export function useContributionsReminder(
+    reminderPeriod: string,
+    reminderPlatform: ReminderPlatform,
+    reminderComponent: ReminderComponent,
+    reminderStage: ReminderStage,
+): ContributionsReminder {
```

Comment:
Maybe the naming could be improved here? I was struggling to come up with a good one. `useContributionsReminderSignup`?

Reply:
`useContributionsReminderSignup` sounds right